### PR TITLE
[CDAP-1400] Removes OperationException completely from our code base

### PR DESCRIPTION
**Pull request by anew**
my weekly tech debt payoff.

This exception was only thrown in a few places in the metrics system:
- for illegal increments in HBase metrics table implementation. Changed that to throw NumberFormatException as in the other two implementations of MetricsTable.
- for a bunch of methods in watchdog that used it to wrap any exception that occurred. Changed these places to throw the original exceptions and also changed the catch clauses of callers accordingly.
- Nowhere in the code did we check the status code of an OperationException, or did anything other than passing through to the caller, or log and ignore. 
- StatusCode was also removed as it was only used by this exception.

### CODE REVIEW

File: cdap-watchdog-tests/src/test/java/co/cask/cdap/metrics/data/AggregatesTableTestBase.java [hunk 1]
```diff
@@ -470,39 +469,6 @@ public void testDeletes() throws OperationException {
     }
   }
 
-  @Test
-  public void testSwap() throws OperationException {
```
Comment (chtyim) [inline cdap-watchdog-tests/src/test/java/co/cask/cdap/metrics/data/AggregatesTableTestBase.java]: Why removed?

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/data/AggregatesTable.java [hunk 2]
```diff
@@ -59,76 +57,47 @@
...
-   *
-   * @param context the context of the metric.
-   * @param metric the name of the metric.
-   * @param runId the runid of the metric.
-   * @param tag the tag of the metric.  If null, the empty tag is used.
-   * @param oldValue the expected value of the column. If null, this means that the column must not exist.
-   * @param newValue the new value of the column. If null, the effect to delete the column if the comparison succeeds.
-   * @return whether the write happened, that is, whether the existing value of the column matched the expected value.
-   */
-  public boolean swap(String context, String metric, String runId, String tag, Long oldValue, Long newValue)
```
Comment (chtyim) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/data/AggregatesTable.java]: So we don't need this method anymore?? Seems unrelated to the refactoring.

**Comment by anew** [comment]
building here: https://builds.cask.co/browse/CDAP-DUT701

File: cdap-watchdog-tests/src/test/java/co/cask/cdap/metrics/data/AggregatesTableTestBase.java (hunk 1 above)
Comment (anew) [inline cdap-watchdog-tests/src/test/java/co/cask/cdap/metrics/data/AggregatesTableTestBase.java]: it was nowhere used except in tests

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/data/AggregatesTable.java (hunk 2 above)
Comment (anew) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/data/AggregatesTable.java]: True, unrelated but it was affected by the refactor. Better to remove than to refactor.

**Comment by chtyim** [comment]
LGTM
